src/sql/base/query/browser/untitledQueryEditorInput.ts

```diff
@@ -44,6 +48,26 @@ export class UntitledQueryEditorInput extends QueryEditorInput implements IUntit
 		return this.text.model.hasAssociatedFilePath;
 	}
 
+	override async save(group: GroupIdentifier, options?: ISaveOptions): Promise<IEditorInput | undefined> {
+		let newEditorInput = await this.text.save(group, options);
+		return this.saveProcess(newEditorInput);
+
+	}
+
+	override async saveAs(group: GroupIdentifier, options?: ISaveOptions): Promise<IEditorInput | undefined> {
+		let newEditorInput = await this.text.saveAs(group, options);
```

Comment:
Please add comments here explaining why we need to manually create the file input. Something like 

```
Create our own FileQueryEditorInput wrapper here so that the existing state (connection, results, etc) can be transferred from this untitled input to the new file input.
```